spot-ml/src/main/scala/org/apache/spot/netflow/model/FlowSuspiciousConnectsModel.scala

```diff
@@ -58,13 +58,17 @@ class FlowSuspiciousConnectsModel(topicCount: Int,
     val wordToPerTopicProbBC = sc.broadcast(wordToPerTopicProb)
 
 
+    /** A left outer join (below) takes rows from the left DF for which the join expression is not
+      * satisfied (for any entry in the right DF), and fills in 'null' values (for the additional columns).
+      */
     val dataWithSrcTopicMix = {
-      val joinedDF = inDF.join(ipToTopicMixDF, inDF(SourceIP) === ipToTopicMixDF("ip"))
+      val joinedDF = inDF.join(ipToTopicMixDF, inDF(SourceIP) === ipToTopicMixDF("ip"), "left_outer")
```

Comment:
I think I named these variables, so it's not your fault Brandon...
but these variable names stink.   I am sorry. A variable name should tell us  what something is, not how it was computed (as that is redundant with the code... that is, "joinedDF is the dataframe you get by taking in the in dataframe and joining it with something else"

I suggest:

recordsWithSrcIPTopicMixes for joinedDF

and then

recordsWithTopicMixes for joinedDF2

sorry about that